### PR TITLE
Fix NPE of cumulative ack mode and incorrect unack message count

### PR DESCRIPTION
**Pull request by congbobo184**
link https://github.com/apache/pulsar/pull/13383
## Motivation
#13383 has fixed  the batch message ack does not decrease the unacked-msg count, but in cumulative ack mode also decrease, it will use pendingAcks, but in cumulative ack, this will not init.

image
image

If ack the batch index one by one, the last ack of a batch will decrease unack message with `batchSize`
```
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 1
================ batch size -> 10
================ message id -> 3:1
================ acked count -> 9
================ batch size -> 10
```

### Modifications
add judge `Subscription.isIndividualAckMode(subType)` when get ackCount.
If the ack from consumer don't have ackset, we should treat it as empty ackset to calculate the ack count with the currently ackset.

### Verifying this change
Add the tests for it

Does this pull request potentially affect one of the following parts:
If yes was chosen, please highlight the changes

Dependencies (does it add or upgrade a dependency): (no)
The public API: (no)
The schema: (no)
The default values of configurations: (no)
The wire protocol: (no)
The rest endpoints: (no)
The admin cli options: (no)
Anything that affects deployment: (no)

### CODE REVIEW

**Review by mattisonchao — APPROVED** [review]

**Comment by mattisonchao** [comment]
@Technoboy-   PTAL :)

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
Unack messages for Failover and Exclusive subscription is meaningless.

**Review by eolivelli — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java [hunk 1]
```diff
@@ -519,9 +521,10 @@ private long getBatchSize(MessageIdData msgId) {
         return batchSize;
     }
 
-    private long getAckedCountForBatchIndexLevelEnabled(PositionImpl position, long batchSize, long[] ackSets) {
+    private long getAckedCountForBatchIndexLevelEnabled(PositionImpl position, long batchSize,
+                                                        long[] ackSets, SubType subType) {
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:525]: Seems that we don't need `subType`.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java [hunk 2]
```diff
@@ -519,9 +521,10 @@ private long getBatchSize(MessageIdData msgId) {
...
         long ackedCount = 0;
-        if (isAcknowledgmentAtBatchIndexLevelEnabled) {
+        if (Subscription.isIndividualAckMode(subType) && isAcknowledgmentAtBatchIndexLevelEnabled) {
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java:527]: Suggest to change to  if (isAcknowledgmentAtBatchIndexLevelEnabled && Subscription.isIndividualAckMode(subType) )

**Pull request merged**